lib/workers/repository/update/branch/bump-versions.ts

```diff
@@ -0,0 +1,269 @@
+import { type ReleaseType, inc } from 'semver';
+import type { BumpVersionConfig } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { scm } from '../../../../modules/platform/scm';
+import { coerceArray } from '../../../../util/array';
+import { readLocalFile } from '../../../../util/fs';
+import type { FileChange } from '../../../../util/git/types';
+import { regEx } from '../../../../util/regex';
+import { matchRegexOrGlobList } from '../../../../util/string-match';
+import { compile } from '../../../../util/template';
+import type { BranchConfig } from '../../../types';
+import { getFilteredFileList } from '../../extract/file-match';
+
+type ParseFileChangesResult =
+  | { state: 'modified'; content: string | null }
+  | { state: 'deleted' }
+  | { state: 'unmodified' };
+
+export async function bumpVersions(config: BranchConfig): Promise<void> {
+  const bumpVersions = config.bumpVersions;
+  if (!bumpVersions?.length) {
+    return;
+  }
+
+  // skip if no packageFiles or artifacts have been updated
+  if (!config.updatedPackageFiles?.length && !config.updatedArtifacts?.length) {
+    return;
+  }
+
+  const allFiles = await scm.getFileList();
+  const fileList = getFilteredFileList(config, allFiles);
+
+  const packageFileChanges = fileChangeListToMap(config.updatedPackageFiles);
+  const artifactFileChanges = fileChangeListToMap(config.updatedArtifacts);
+
+  for (const bumpVersionConfig of bumpVersions) {
+    await bumpVersion(
+      bumpVersionConfig,
+      config,
+      fileList,
+      packageFileChanges,
+      artifactFileChanges,
+    );
+  }
+
+  // update the config with the new files
+  config.updatedPackageFiles = Object.values(packageFileChanges).flat();
+  config.updatedArtifacts = Object.values(artifactFileChanges).flat();
+}
+
+async function bumpVersion(
+  config: BumpVersionConfig,
+  branchConfig: BranchConfig,
+  fileList: string[],
+  packageFiles: Record<string, FileChange[]>,
+  artifactFiles: Record<string, FileChange[]>,
+): Promise<void> {
+  const bumpVersionsDescr = config.name
+    ? `bumpVersions(${config.name})`
+    : 'bumpVersions';
+
+  const files: string[] = [];
+  try {
+    files.push(...getMatchedFiles(config.filePatterns, branchConfig, fileList));
+  } catch (e) {
+    addArtifactError(
+      branchConfig,
+      `Failed to calculate matched files for bumpVersions: ${e.message}`,
+    );
```

Comment:
```suggestion
    );
    return;
```

makes no sense to move on without files